fava_investor/modules/performance/balances.py

```diff
@@ -0,0 +1,77 @@
+import datetime
+from typing import Dict, List, Union
+
+from beancount.core import realization
+from beancount.core.data import iter_entry_dates
+from fava.core import FavaLedger
+from util import pairwise
+from util.date import Interval
+
+ConfigDict = Dict[str, Union[str, List[str]]]
+
+
+def filter_nested_accounts_dict(accounts_dict: Dict[str, Dict],
+                                accounts_to_keep: List[str], prev:str = "") -> dict:
+    """
+    filters accounts_dict of following format:
+    {
+        Assets: {
+            Bank: {
+                BankA: {...},
+                ... },
+            Investment: {
+                BrokerA: {...},
+                ... }
+            ... }
+        ... }
+    """
+    for key in list(accounts_dict.keys()):
+        str_begin = f"{prev}{key}"
+        if str_begin not in [acc[:len(str_begin)] for acc in accounts_to_keep]:
+            del accounts_dict[key]
+        else:
+            accounts_dict[key] = filter_nested_accounts_dict(accounts_dict[key], accounts_to_keep, prev + key + ":")
```

Comment:
We should avoid recursive functions. They make code hard to read/maintain and also prone to stack errors.

This function may not be needed at all. Can you use realization.filter instead?

Reply:
I reworked the module to build a `Tree` instead of previous structures and filter out from non-value accounts. I can't see other way to remove accounts from tree other than recursively. Do you have any suggestions?